src/Processors/StreamingAdapter.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <Poco/Logger.h>
+
+#include <Processors/IProcessor.h>
+
+namespace DB
+{
+
+class StreamingAdapter final : public IProcessor
```

Comment:
I think yes, for now it can be replaced to ConcatProcessor, idea behind this class to make chunk reordering with respect to their cursors.

Proposed solution in issue operates with cursors - a map with stream-name -> offset.

Idea of how to manage storage cursors: If storage supports cursors, it should set up cursor for each chunk generated from source and in sink it also should set up cursor for a chunk before pushing it to subscribers. Cursor can be stored in ChunkInfo for example. When chunks are being consumed by StreamingAdapter it will sort them accordingly to their offsets inside this map. As far as I can tell, chunks at this part of pipeline could not contain multiple stream-name in cursor, because stream-name ~= partition-id. 

tldr: after this processor storage must be visible as FIFO Queue.

But this is just an idea.